### DISCUSSION TITLE
How do you add `compose-compiler` to kotlin monorepo

### DISCUSSION THREAD

**Discussion by ede-wolf-gls** [renovatebot/renovate#19082] [Request Help] [answered]
The androidx compose compiler version is tightly bound to the kotlin version ([compatibility page](https://developer.android.com/jetpack/androidx/releases/compose-kotlin)). So it would be nice to upgrade both in the same PR.
So my approach is to add the compose compiler dependency to the kotlin monorepo preset. But I configured it wrong or this is all in all the wrong approach for it. Could someone give me a hint, how I can configure this?
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchSourceUrlPrefixes": ["https://github.com/JetBrains/kotlin"],
      "matchPackageNames": ["androidx.compose.compiler:compiler"],
      "groupName": "kotlin monorepo"
    }
  ]
}
```

**Comment by rarkins** [comment, marked as answer]
These matching are treated like "AND" so you should instead separate into two rules:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchSourceUrlPrefixes": ["https://github.com/JetBrains/kotlin"],
      "groupName": "kotlin monorepo"
    },     {
      "matchPackageNames": ["androidx.compose.compiler:compiler"],
      "groupName": "kotlin monorepo"
    }
  ]
}
```
